lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp

```diff
@@ -341,5 +341,34 @@ template std::optional<Value> getConstTensor<int64_t>(PatternRewriter &,
                                                       Operation *,
                                                       ArrayRef<int64_t> vec,
                                                       ArrayRef<int64_t> shape);
+
+LogicalResult getAvgPool2dAccType(PatternRewriter &rewriter, Value input,
+                                  TypeAttr &accType) {
+  auto inputTy = llvm::dyn_cast<ShapedType>(input.getType());
+  if (!inputTy)
+    return failure();
+  auto inputETy = inputTy.getElementType();
```

Comment:
(Background)
The goal with the change we made to TOSA was to allow either fp32 or fp16 accumulation for average pool to trade off speed/range of intermediate value. The accumulator size isn't visible outside the op, so it needs this new attribute. At first we're leaving accumulator as always fp32 while finding the right time/method to choose fp16.

For legalizing from torch, there is an easier way than what the validator does. You can reduce the type choice to something like:
```
  // Tosa supports FP16 and FP32 accumulator type for FP16 input. When the time
  // FP16 is supported, the accumulator type can be selected based on trade-off
  // between performance and accuracy. Set to FP32 by default.
 accType = inputETy.isa<FloatType>()
                          ? mlir::TypeAttr::get(rewriter.getF32Type())
                          : mlir::TypeAttr::get(rewriter.getIntegerType(32));
```